src/node.cpp

```diff
@@ -243,16 +244,37 @@ Status Node::demultiplyOutputs(SessionResults& nodeSessionOutputs) {
         const auto step = blob->byteSize() / resultsDemultiplyCount;
         for (size_t i = 0; i < newSessionMetadatas.size(); ++i) {
             InferenceEngine::Blob::Ptr dividedBlob;
-            auto status = createSharedBlob(dividedBlob, dividedBlobDesc);
-            if (!status.ok()) {
-                return status;
-            }
-            if (dividedBlob->byteSize() != step) {
-                SPDLOG_LOGGER_ERROR(dag_executor_logger, "Node: {}, session: {} created blob: {} have wrong byte size: {}, expected: {}",
-                    getName(), metadata.getSessionKey(), blobName, dividedBlob->byteSize(), step);
-                return StatusCode::UNKNOWN_ERROR;
+            // if condition is perf optimization
+            // when demultiplying from entry node from tensor content we can skip allocation for sharded blobs
+            // and reuse memory from original blob since its memory is kept for whole duration of predict request
+            if (getName() == ENTRY_NODE_NAME) {
+                if (tensorDesc.getPrecision() == InferenceEngine::Precision::FP32)
+                    dividedBlob = InferenceEngine::make_shared_blob<float>(dividedBlobDesc, (float*)blob->buffer() + i * step / sizeof(float));
+                if (tensorDesc.getPrecision() == InferenceEngine::Precision::U8)
+                    dividedBlob = InferenceEngine::make_shared_blob<uint8_t>(dividedBlobDesc, (uint8_t*)blob->buffer() + i * step / sizeof(uint8_t));
+                if (tensorDesc.getPrecision() == InferenceEngine::Precision::I8)
+                    dividedBlob = InferenceEngine::make_shared_blob<int8_t>(dividedBlobDesc, (int8_t*)blob->buffer() + i * step / sizeof(int8_t));
+                if (tensorDesc.getPrecision() == InferenceEngine::Precision::I16)
```

Comment:
Can we separate this blob creation method to ov_utils.hpp? We already tried doing that at some point, maybe we can follow this idea here.

Reply:
We agreed to leave it here since the logic behind depends on entry node implementation. This part will be moved to entry 
node